kafka-client/src/main/java/dev/responsive/kafka/internal/utils/TableName.java

```diff
@@ -18,38 +18,48 @@
 
 import java.util.Objects;
 import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * {@code TableName} represents the name for a table and all variations
  * of it - some characters that are valid for Kafka names are not valid
- * as Cassandra table names.
+ * as table names.
  */
 public class TableName {
 
-  // cassandra names are case-insensitive and can only contain
+  private static final Logger LOG = LoggerFactory.getLogger(TableName.class);
+
+  private static final int MAXIMUM_LENGTH = 65;
+
+  // table names are case-insensitive and should only contain
   // alphanumeric and underscore characters
-  private static final Pattern INVALID_CASSANDRA_CHARS =
+  private static final Pattern INVALID_CHARS =
       Pattern.compile("[^a-zA-Z0-9_]");
 
   private final String kafkaName;
-  private final String remoteName;
+  private final String tableName;
 
   public TableName(final String kafkaName) {
     this.kafkaName = kafkaName;
     final var escaped = kafkaName.replaceAll("_", "__"); // escape existing underscores
-    remoteName = INVALID_CASSANDRA_CHARS
+    tableName = INVALID_CHARS
         .matcher(escaped)
         .replaceAll("_")
         .toLowerCase();
+
+    if (tableName.length() > MAXIMUM_LENGTH) {
+      LOG.error("Invalid table name for state store {}, must be no more than {} characters long"
```

Comment:
FYI: I do intend to eventually throw an error here, but some of our tests are exceeding this limit. I'm a bit confused because from the docs it sounds like C* has, if anything, a lower character count limit on table names...but it might be that the limits I'm looking at are only for Scylla, and the embedded C* we use in the test harness doesn't have such a limit. Hence things passing now

Frankly it would be kind of a bummer to have to limit all our test names to 65 characters or less, but we'll have to do so for anything using a Mongo backend. And since we should probably parametrize the tests to use both C* and Mongo in the end, the only thing we'd be restricting "unnecessarily" are the Cassandra-specific unit tests.

Personally I'm down to just rename them, but it's certainly a bit annoying. Any strong feelings on this before I go ahead?

Reply:
Oh actually, the tests I thought were too long were just timing out for other reasons. 65 characters is definitely still a bit of a constraint, but it leaves a fair amount of room. I'm going to go ahead with throwing the error here